File: legacy/sdk/python/axion/resources/tasks.py
"""Tasks resource."""
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from axion.client import Axion


class TasksResource:
    def __init__(self, client: "Axion") -> None:
        self._c = client

    def list(
        self,
        *,
        status: str | None = None,
        agent_id: str | None = None,
        workflow_run_id: str | None = None,
        limit: int = 50,
        offset: int = 0,
    ) -> list[dict]:
        params = {
            k: v
            for k, v in {
                "status": status,
                "agent_id": agent_id,
                "workflow_run_id": workflow_run_id,
                "limit": limit,
                "offset": offset,
            }.items()
            if v is not None
        }
        return self._c._request("GET", "/tasks", params=params)

    def get(self, task_id: str) -> dict:
        return self._c._request("GET", f"/tasks/{task_id}")

    def cancel(self, task_id: str) -> None:
        self._c._request("POST", f"/tasks/{task_id}/cancel")

File: legacy/sdk/typescript/src/resources/tasks.ts
/** Tasks resource. */
import type { Axion } from "../client.js";
import type { Task } from "../types.js";

export class TasksResource {
  constructor(private readonly _c: Axion) {}

  list(filter: {
    status?: string;
    agentId?: string;
    workflowRunId?: string;
    limit?: number;
    offset?: number;
  } = {}): Promise<Task[]> {
    return this._c._request<Task[]>("GET", "/tasks", {
      query: {
        status: filter.status,
        agent_id: filter.agentId,
        workflow_run_id: filter.workflowRunId,
        limit: filter.limit,
        offset: filter.offset,
      },
    });
  }

  get(taskId: string): Promise<Task> {
    return this._c._request<Task>("GET", `/tasks/${taskId}`);
  }

  async cancel(taskId: string): Promise<void> {
    await this._c._request("POST", `/tasks/${taskId}/cancel`);
  }
}
